### PR TITLE
Use Github Actions

### PR DESCRIPTION
**Pull request by curiositycasualty**
This mainly does 2 things:
  - swaps to Github Actions for CI (instead of travis) and
  - gussies up the existing formulae to meet `brew style` and `brew audit` standards

If `brew style` and `brew audit` are good enough for homebrew-core, they're good enough for your tap.

### CODE REVIEW

**Comment by curiositycasualty** [comment]
Looks like I've got something to fix with regards to the `resty` CLI. Please stand by.

**Comment by curiositycasualty** [comment]
Took me way too long to figure out that converting the `openresty` formulae to `keg-only` (as `brew style`/`brew audit` instructed me to), causes the symlinks for the `rest`/`nginx` binaries to NOT be created as they were previously.
This led to the formulae installing successfully, but is causing the `kong` CLI tool to fail:
```
env: resty: No such file or directory
Error: Process completed with exit code 127.
```

As seen here: https://github.com/Kong/homebrew-kong/runs/4537975626?check_suite_focus=true#step:12:5374

This log message when installing `openresty` directly tipped me off:
```
==> Caveats
openresty@1.19.3.2 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

If you need to have openresty@1.19.3.2 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/openresty@1.19.3.2/bin:$PATH"' >> ~/.zshrc
```

Let me figure out if there is either a way to not need to be keg-only without pissing off `brew audit`, or if there is a way to force the binary installs for keg-only fomulae.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      - "Formula/*.rb"
+    branches:
+      - master
+
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test Formulae
+    runs-on: macos-11
```
Comment (mayocream) [inline .github/workflows/ci.yml:21]: Can we test on multiple vms? like this:
https://github.com/Homebrew/homebrew-core/blob/6287ae3017b91a518347e69c8587aa9e9b6d5a64/.github/workflows/tests.yml#L183-L191

Comment (curiositycasualty) [reply .github/workflows/ci.yml:21]: We can.. but would take some extra work I don't want to put into this repo just yet.

That would be accomplished via GH actions' `matrix`/`strategy` syntax: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-running-multiple-versions-of-nodejs

The list of platforms available are here: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

Notably lacking are Mac OS arm64/M1 processor architecture support. See also: https://github.com/actions/virtual-environments/issues/2187

**Review by locao — APPROVED** [review]

**Pull request merged**